python/ray/util/client/common.py

```diff
@@ -412,6 +420,26 @@ def _get_client_id_from_context(context: Any, logger: logging.Logger) -> str:
     return client_id
 
 
+def _propogate_error_in_context(e: Exception, context: Any) -> bool:
+    """
+    Encode an error into the context of an RPC response. Returns True
+    if the error can be recovered from, false otherwise
+    """
+    try:
+        if isinstance(e, grpc.RpcError):
+            # RPC error, propagate directly by copying details into context
+            context.set_code(e.code())
+            context.set_details(e.details())
+            return e.code() not in GRPC_UNRECOVERABLE_ERRORS
+    except Exception:
+        # Extra precaution -- if encoding the RPC directly fails fallback
+        # to treating it as a regular error
+        pass
+    context.set_code(grpc.StatusCode.FAILED_PRECONDITION)
```

Comment:
Is there any logic depending on this status code?

Reply:
No, just has to be one of the codes that are reserved for application failures, and not ABORTED (reserved encoding for pickle.dumps'd errors)